packages/block-editor/src/store/selectors.js

```diff
@@ -1388,6 +1389,101 @@ export function canInsertBlocks( state, clientIds, rootClientId = null ) {
 	);
 }
 
+/**
+ * Determines if the given block type is allowed to be deleted.
+ *
+ * @param {Object}         state        Editor state.
+ * @param {string|Object}  blockName    The block type object, e.g., the response
+ *                                      from the block directory; or a string name of
+ *                                      an installed block type, e.g.' core/paragraph'.
+ * @param {?string}        rootClientId Optional root client ID of block list.
+ *
+ * @return {boolean} Whether the given block type is allowed to be removed.
+ */
+export function canRemoveBlockType( state, blockName, rootClientId = null ) {
+	let blockType;
+	if ( blockName && 'object' === typeof blockName ) {
+		blockType = blockName;
+		blockName = blockType.name;
+	} else {
+		blockType = getBlockType( blockName );
+	}
+	// undefined blocks can still be deleted.
+	if ( ! blockType ) {
+		return true;
+	}
```

Comment:
My assumption here was that an undefined block type means the block is not registered, but it's still a block, so it can be removed.
However, this ignores if the parent is locked or not. I can remove this and make all undefined blocks removable but it feels more like restrictions.

Reply:
Seeing this, all of this block (1404 to 1414) can be removed, it would always differ to the parent lock status.